File: src/types/default-model.ts
export type DefaultModelKey = 'assistant' | 'quick' | 'translation' | 'embedding';

export interface DefaultModelSetting {
  key: DefaultModelKey;
  provider_id: string;
  model_id: string;
  temperature: string;
  top_p: string;
  max_tokens: string;
  presence_penalty: string;
  frequency_penalty: string;
}

File: src-tauri/src/models/default_model.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct DefaultModelSetting {
    pub key: String,
    pub provider_id: String,
    pub model_id: String,
    #[serde(default)]
    pub temperature: String,
    #[serde(default)]
    pub top_p: String,
    #[serde(default)]
    pub max_tokens: String,
    #[serde(default)]
    pub presence_penalty: String,
    #[serde(default)]
    pub frequency_penalty: String,
}
